### PR TITLE
Improve error message when Xcode is not found

### PR DESCRIPTION
**Pull request by premun**
This improves the experience when `xcode-select` points somewhere bad (like xcode command tools).

https://github.com/dotnet/xharness/issues/337

### CODE REVIEW

**Comment by premun** [comment]
This steps was failing because one of the machines didn't have `xcode-select` pointing to the Xcode installation but to the Xcode command line tools. That's why it only failed occasionally when it hit this one machine.

I verified that this machine was the only one with this issue:
```sql
WorkItems
| where QueueName == "osx.1015.amd64.open" and Started > now()-30d and FriendlyName == "xharness-ios-state" and Status != "Pass"
```

Error message improved here https://github.com/dotnet/xharness/pull/349

**Review by MattGal — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.Common/Execution/MacOSProcessManager.cs [hunk 1]
```diff
@@ -40,8 +40,19 @@ public Version XcodeVersion
...
+                    try
+                    {
+                        doc.Load(plistPath);
+                        _xcode_version = Version.Parse(doc.SelectSingleNode("//key[text() = 'CFBundleShortVersionString']/following-sibling::string").InnerText);
+                    }
+                    catch (IOException)
+                    {
+                        throw new Exception(
+                            $"Failed to find Xcode! Version.plist missing at {plistPath}. " +
+                            $"Please make sure xcode-select is set up or the path to Xcode is suplied as argument.");
```
Comment (MattGal) [inline src/Microsoft.DotNet.XHarness.Common/Execution/MacOSProcessManager.cs:54]: nit:  not an interpolated string

Comment (premun) [reply src/Microsoft.DotNet.XHarness.Common/Execution/MacOSProcessManager.cs:54]:
```suggestion
                            "Please make sure xcode-select is set up or the path to Xcode is suplied as argument.");
```

File: src/Microsoft.DotNet.XHarness.Common/Execution/MacOSProcessManager.cs [hunk 2]
```diff
@@ -40,8 +40,19 @@ public Version XcodeVersion
...
+                    try
+                    {
+                        doc.Load(plistPath);
+                        _xcode_version = Version.Parse(doc.SelectSingleNode("//key[text() = 'CFBundleShortVersionString']/following-sibling::string").InnerText);
+                    }
+                    catch (IOException)
+                    {
+                        throw new Exception(
+                            $"Failed to find Xcode! Version.plist missing at {plistPath}. " +
+                            "Please make sure xcode-select is set up or the path to Xcode is suplied as argument.");
```
Comment (premun) [inline src/Microsoft.DotNet.XHarness.Common/Execution/MacOSProcessManager.cs:54]:
```suggestion
                            "Please make sure xcode-select is set up, or the path to Xcode is supplied as an argument.");
```

**Pull request merged**